### PR TITLE
Online TempoDetector: (Comb, ACF, DBN)

### PR DESCRIPTION
**Pull request by SebastianPoell**
Hey,

Here my first Pull Request. This should work now (inspired by the online version of the DBNBeatTracker):

python3 TempoDetector -v online
python3 TempoDetector online /wathever.wav
python3 TempoDetector single --online /whatever.wav

Update: Also added support for online acf and dbn:
python3 TempoDetector --method acf -v online
python3 TempoDetector --method dbn -v online

Let me know if there are things to change :)

Regards, Sebi

### CODE REVIEW

File: madmom/features/beats.py [hunk 1]
```diff
@@ -1147,7 +1147,7 @@ def process_forward(self, activations, reset=True, **kwargs):
             sys.stderr.write('\r%s' % ''.join(display))
             sys.stderr.flush()
         # forward path often reports multiple beats close together, thus report
-        # only beats more than the minumum interval apart
+        # only beats more than the minimum interval apart
```
Comment (superbock) [inline madmom/features/beats.py]: Please bundle these two fixes in a single commit with a meaningful commit message, e.g. 'fix typos'.

File: madmom/features/tempo.py [hunk 2]
```diff
@@ -215,7 +217,7 @@ def detect_tempo(histogram, fps):
     if len(peaks) == 0:
         # a flat histogram has no peaks, use the center bin
         if len(bins):
-            ret = np.asarray([tempi[len(bins) / 2], 1.])
+            ret = np.asarray([tempi[int(len(bins) / 2)], 1.])
```
Comment (superbock) [inline madmom/features/tempo.py]: Nice catch! Already included in #292.

File: madmom/features/tempo.py [hunk 3]
```diff
@@ -512,6 +726,7 @@ def write_tempo(tempi, filename, mirex=False):
     # format as a numpy array
     out = np.array([t1, t2, strength], ndmin=2)
     # write to output
-    np.savetxt(filename, out, fmt='%.2f\t%.2f\t%.2f')
+    if filename is not None:
+        np.savetxt(filename, out, fmt='%.2f\t%.2f\t%.2f')
```
Comment (superbock) [inline madmom/features/tempo.py]: Am I missing something here? How does this change add any meaningful functionality?
Please revert the changes in lines 681-730 or convince me that this is really needed. Line 692 needs to go away nonetheless.

Comment (SebastianPoell) [inline madmom/features/tempo.py]: Ah yeah, I thought for the **verbose** online mode it would be nice to print the current most likely tempi and their relative strength to the console - just like it's done in the result files (tempo_1, tempo_2, strength). Since the write_tempo method not only prints those values into a file but also returns them I ended up using this for displaying the current estimation in the console:

screen shot 2017-06-06 at 07 41 23

For writing it to the console no writing to a textfile is needed (that's where the line of code comes from). Should I change this somehow?

Comment (superbock) [inline madmom/features/tempo.py]: Isn't the visualize stuff [in line 411+ in tempo.py exactly for this](https://github.com/SebastianPoell/madmom/blob/master/madmom/features/tempo.py#L411)? At least it is how [I did it for beats](https://github.com/CPJKU/madmom/blob/master/madmom/features/beats.py#L1124).

**Comment by SebastianPoell** [comment]
I will close that PR now since it will be refactored in #294 ;)
